presentations/async-await-intro/5.rs

```diff
@@ -0,0 +1,13 @@
+async fn call_some_api() {}
+
+async fn do_interesting_things() {
+    println!("I'm an async function and I do interesting things");
+    call_some_api().await
+}
+
+fn main() {
+    do_interesting_things(); // <1>
+    async {
+        do_interesting_things(); // <2>
+    };
```

Comment:
As a first example, we should maybe show them `.await` vs not having `.await`. `async` blocks could come later.